### PR TITLE
Console functions

### PR DESCRIPTION
**Pull request by hazzard993**
For #439

Added some functions from `console` that translate to Lua. `log`, `assert` and `trace`.

```ts
console.log("Hello");               // print("Hello")
console.log("Hello %s", "there");   // prints "Hello there" in Lua
console.log("Hello", "There");      // print("Hello", "There")
console.assert(true);               // assert(true)
console.assert(true, "message");    // assert(true, "message")
console.assert(false, "Hello %s", "there"); // Fails with "Hello there" as a message
console.trace();                    // Prints a stack trace in Lua
console.trace("message");           // Prints a stack trace with "message" at the top in Lua
console.trace("Hello %s", "there"); // Prints a stack trace with "Hello there" at the top in Lua
```

### CODE REVIEW

**Comment by tomblind** [comment]
Will this work?
```ts
console.log("foo%%bar", "blah"); // Should print "foo%bar blah"
```
Also, any thoughts on console.warn and console.error?

File: test/unit/console.spec.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @TestCase('console.log("Hello", "There")', 'print("Hello", "There");')
+    @TestCase("console.trace()", "print(debug.traceback());")
+    @TestCase('console.trace("message")', 'print(debug.traceback("message"));')
+    @TestCase('console.trace("Hello %s", "there")', 'print(debug.traceback(string.format("Hello %s", "there")));')
+    @TestCase('console.trace("Hello", "there")', 'print(debug.traceback("Hello", "there"));')
+    @TestCase("console.assert(false)", "assert(false);")
+    @TestCase('console.assert(false, "message")', 'assert(false, "message");')
+    @TestCase('console.assert(false, "message %s", "info")', 'assert(false, string.format("message %s", "info"));')
+    @TestCase('console.assert(false, "message", "more")', 'assert(false, "message", "more");')
+    @Test("Console")
```
Comment (Perryvw) [inline test/unit/console.spec.ts]: You can split these up into different test cases: one for log, one for trace, one for assert, and put the input parameters in the TestCase (i.e. `@TestCase("")`, `@TestCase("Hello %")`)

File: src/LuaTransformer.ts [hunk 2]
```diff
@@ -3544,6 +3543,90 @@ export class LuaTransformer {
...
     }
 
+    public transformConsoleCallExpression(expression: ts.CallExpression): ExpressionVisitResult {
+        const method = expression.expression as ts.PropertyAccessExpression;
+        const methodName = method.name.escapedText;
+
+        switch (methodName) {
+            case "log":
+                if (expression.arguments.length > 0) {
+                    if (ts.isStringLiteral(expression.arguments[0])
```
Comment (Perryvw) [inline src/LuaTransformer.ts]: It's probably a good idea to refactor this to a private helper method under this function:
```ts
private isStringFormatTemplate(expression: ts.Expression): boolean {
    return ts.isStringLiteral(expression) && expression.text.includes("%");
}
```

**Comment by hazzard993** [comment]
Welp, this has gone bad. I'm gunna try again

**Pull request closed without merging**